module/src/link.rs

```diff
@@ -65,28 +65,49 @@ pub trait Linkable: Send + Sync {
 /// [`Linkable`]: ./trait.Linkable.html
 /// [`send`]: ./trait.Port.html#tymnethod.send
 /// [`receive`]: ./trait.Port.html#tymnethod.receive
-pub trait Port: Any + 'static {
-    /// Sets to send a list of handles to the other end on link.
-    ///
-    /// `desc` is encoded in `CBOR`.
-    fn send(&mut self, desc: &[u8]) -> &mut dyn Port;
+pub trait Port: CastFrom {
+    /// Sets to send a list of handles represented by the `ids` to the other end on link
+    /// creation. The `ids` are indices into a list of service objects created when the module
+    /// owning this port is loaded into a sandbox.CBOR map fed
+    /// to the constructor function.
+    fn export(&mut self, ids: &[usize]) -> &mut dyn Port;
```

Comment:
Also, some module might want to provide an opaque argument `Vec<u8>` for export/importing to provide various ways of interpreting the `ids`. I'm not sure but are you trying to provide such complexity with `load()`?

Reply:
`export` is just about what will be passed into the other side of the link. Link implementations are free to transfer additional data internally on link creation time.